prime-angular-frontend/src/app/modules/enrolment/pages/regulatory/regulatory.component.ts

```diff
@@ -0,0 +1,156 @@
+  templateUrl: './regulatory.component.html',
+  styleUrls: ['./regulatory.component.scss']
+})
+export class RegulatoryComponent implements OnInit, OnDestroy {
+
+  public form: FormGroup;
+  public jobForm: FormGroup;
+  public colleges: Config<number>[];
+  public licenses: Config<number>[];
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private dialog: MatDialog,
+    private configService: ConfigService,
+    private enrolmentStateService: EnrolmentStateService,
+    private enrolmentResource: EnrolmentResource,
+    private toastService: ToastService,
+    private logger: LoggerService
+  ) {
+    this.colleges = this.configService.colleges;
+    this.licenses = this.configService.licenses;
+  }
+
+  public get certifications(): FormArray {
+    return this.form.get('certifications') as FormArray;
+  }
+
+  public onSubmit() {
+    if (this.form.valid) {
+      this.clearEmptyCertifications();
+      this.clearJobForm();
+      const payload = this.enrolmentStateService.enrolment;
+      this.enrolmentResource.updateEnrolment(payload)
+        .subscribe(
+          () => {
+            this.toastService.openSuccessToast('Regulatory information has been saved');
+            this.form.markAsPristine();
+            this.certifications.clear();
+            this.router.navigate(['device-provider'], { relativeTo: this.route.parent });
+          },
+          (error: any) => {
+            this.toastService.openErrorToast('Regulatory information could not be saved');
+            this.logger.error('[Enrolment] Regulatory::onSubmit error has occurred: ', error);
+          }
+        );
+      this.form.markAsPristine();
+    } else {
+      this.form.markAllAsTouched();
+    }
+
+
+    // FOR TESTING
+    this.router.navigate(['device-provider'], { relativeTo: this.route.parent });
+  }
+
+  public addCertification() {
+    const certification = this.enrolmentStateService.buildCollegeCertificationForm();
+    this.certifications.push(certification);
+  }
+
+  public removeCertification(index: number) {
+    if (index >= 0) {
+      this.certifications.removeAt(index);
+    }
+  }
+
+  public canDeactivate(): Observable<boolean> | boolean {
+    const data = 'unsaved';
+    return (this.form.dirty)
+      ? this.dialog.open(ConfirmDialogComponent, { data }).afterClosed()
+      : true;
+  }
+
+  public ngOnInit() {
+    this.createFormInstance();
+    // Initialize form changes before patching
+    this.initForm();
+
+    // TODO: detect enrolment already exists and don't reload
+    // TODO: apply guard if not enrolment is found to redirect to profile
+    this.enrolmentResource.enrolments()
+      .subscribe((enrolment: Enrolment) => {
+        if (enrolment) {
+          this.enrolmentStateService.enrolment = enrolment;
+        }
+      });
+  }
+
+  public ngOnDestroy() {
+    this.clearEmptyCertifications();
+  }
+
+  public clearEmptyCertifications() {
+    let i = 0;
```

Comment:
You can shrink down complexity or trim a few lines by chaining methods.  This seems like a candidate for `forEach` on the controls array